README.md

```diff
@@ -175,6 +175,153 @@ Only for now. Once the generator has been published, the Smithy CLI will be able
 to run it without a separate Java installation. Similarly, once the python
 helper libraries have been published you won't need to install them locally.
 
+### How do I generate types for shapes without a client?
+
+If all you want are concrete Python classes for the shapes in your Smithy model,
+all you need to do is replace `python-client-codegen` with
+`python-type-codegen` when following the steps above. Your `smithy-build.json`
+would now look like:
+
+```json
+{
+    "version": "1.0",
+    "sources": ["model"],
+    "maven": {
+        "dependencies": [
+            "software.amazon.smithy:smithy-model:[1.54.0,2.0)",
+            "software.amazon.smithy.python:smithy-python-codegen:0.1.0"
+        ]
+    },
+    "projections": {
+        "shapes": {
+            "plugins": {
+                "python-type-codegen": {
+                    "service": "com.example#EchoService",
+                    "module": "echo",
+                    "moduleVersion": "0.0.1"
+                }
+            }
+        }
+    }
+}
+```
+
+The module with the generated shape classes can be found in
+`build/smithy/client/python-type-codegen` after you run `smithy-build`.
+
+Unlike when generating a client, a service shape is not required for shape
+generation. If a service is not provided then every shape found in the model
+will be generated. Any naming conflicts may be resolved by using the
+[`renameShapes` transform](https://smithy.io/2.0/guides/smithy-build-json.html#renameshapes)
+(or renaming the shapes in the model of course).
+
+The set of shapes generated can also be constrained by using the
+[`includeShapesBySelector` transform](https://smithy.io/2.0/guides/smithy-build-json.html#includeshapesbyselector).
+For example, to generate only shapes within the `com.example` namespace:
+
+```json
+{
+    "version": "1.0",
+    "sources": ["model"],
+    "maven": {
+        "dependencies": [
+            "software.amazon.smithy:smithy-model:[1.54.0,2.0)",
+            "software.amazon.smithy.python:smithy-python-codegen:0.1.0"
+        ]
+    },
+    "projections": {
+        "shapes": {
+            "transforms": [
+                {
+                    "name": "includeShapesBySelector",
+                    "args": {
+                        "selector": "[id|namespace = 'com.example']"
+                    }
+                }
+            ],
+            "plugins": {
+                "python-type-codegen": {
+                    "module": "echo",
+                    "moduleVersion": "0.0.1"
+                }
+            }
+        }
+    }
+}
+```
+
+Input and output shapes (shapes with the `@input` or `@output` traits and
+operation inputs / outputs created as part of an operation definition) are not
+generated by default. To generate these shapes anyway, remove the traits with
+the
+[`excludeTraits` transform](https://smithy.io/2.0/guides/smithy-build-json.html#excludetraits):
```

Comment:
Hmmm.. is this something where there could be a footgun later if they are using a (hypothetical) server plugin or the client plugin? Using type codegen in conjunction with client/server codegen will be common.

Reply:
I'm just calling it out, because, while it would always be possible, I'm not sure we want to point this capability out if it can lead to problems later.

Reply:
Also, Smithy Java generates inputs and outputs without needing the exclusion. We should be consistent.

Reply:
I'd rather they don't generate them tbh. I can make some improvments in this area though

Reply:
I made it a simple toggle in the settings.